### ISSUE TITLE
Your service method does not accept parameters

### ISSUE THREAD

**Issue by serkanagca34** [jonataslaw/getx#2292]
Your service method does not accept parameters. I cannot use the ".obs" property variables that I have created as parameters in my api service methods. and it gives the error **The prefix 'coinTwo' can't be used here because it is shadowed by a local declaration.
Try renaming either the prefix or the local declaration.**  I am sharing the error screenshots and codes. Thanks in advance. @jasonlaw 

`import 'package:get/get.dart';

class BuySaleController extends GetxController {
  RestDataSource api = RestDataSource();

 
  var coinOne = "".obs;
  var coinTwo = "".obs;

  
  var purchase = <Trade>[].obs;
  var sales = <Trade>[].obs;

  @override
  void onInit() {
    getOrderBookData(coinOne.value, coinTwo.value);
    super.onInit();
  }

  //********************** API CALL *********************//

  getOrderBookData(coinOne.value, coinTwo.value) async {
    var res = await api.fetchOrderBookData(coinOne.value, coinTwo.value);
    purchase.value = res.result!.buy!;
    sales.value = res.result!.sell!;
  }
}`

![Untitled](https://user-images.githubusercontent.com/62822922/161737965-f97e7ba2-3b40-4713-ae44-ba72a41ffb62.png)

**Comment by jonataslaw** [comment]
Sorry, I don't understand what you are trying to do. 
In dart, you can pass ONLY Types with variables names by parameter, you are using `coinOne.value` and `coinTwo.value`

**Comment by jonataslaw** [comment]
I think you have trying to write this:
```dart
getOrderBookData(String coinOne, String coinTwo){
  var res = await api.fetchOrderBookData(coinOne, coinTwo);
```

**Comment by serkanagca34** [comment]
yes , but I cannot use the variables I defined above as parameters below. i am getting error.

  var **coinOne** = "".obs;
  var **coinTwo** = "".obs;

`getOrderBookData(String coinOne.value, String coinTwo.value){
  var res = await api.fetchOrderBookData(coinOne, coinTwo);`
